lib/promscrape/scraper.go

```diff
@@ -99,7 +99,8 @@ func WriteConfigData(w io.Writer) {
 
```

Comment:
WDYT. maybe it's more idiomatic to make conditional metrics with set?

```
var (
	configSet          = metrics.NewSet()
	configReloads      = configSet.NewCounter(`vm_promscrape_config_reloads_total`)
	configReloadErrors = configSet.NewCounter(`vm_promscrape_config_reloads_errors_total`)
	configSuccess      = configSet.NewCounter(`vm_promscrape_config_last_reload_successful`)
	configTimestamp    = configSet.NewCounter(`vm_promscrape_config_last_reload_success_timestamp_seconds`)
)

...

if configFile == "" {
		// Nothing to scrape, set configSuccess to 1 in order to avoid false-positive alerts about failed config reloads.
		configSuccess.Set(1)
		return
	}
      metrics.RegisterSet(configSet)
...

```